pymongo/asynchronous/topology.py

```diff
@@ -77,7 +77,7 @@
 _pymongo_dir = str(Path(__file__).parent)
 
 
-def process_events_queue(queue_ref: weakref.ReferenceType[queue.Queue]) -> bool:
+def process_events_queue(queue_ref: weakref.ReferenceType[queue.Queue], closed: bool) -> bool:
```

Comment:
If the executor is sleeping while it's closed, I don't think process_events_queue will every be called.

Reply:
Do we have an existing way to close an executor after it runs its `target` method one more time? Otherwise we'll have to have `Topology.close` just wait until its event queue is empty to close the executor and finish its execution.

Reply:
We discussed in person. We don't have that functionality.